### DISCUSSION TITLE
Sqlalchemy with pytds

### DISCUSSION THREAD

**Discussion by ifuchs** [sqlalchemy/sqlalchemy#8964] [Usage Questions]
I have Jupyter notebook  in which I am able to access using pytds. I would like to use it with sqlalchemy and the %%sql command.

however when try the following:
sqlalchemy.create_engine('mssql+pytds://user:pw@ip/db')
I get:
Engine(mssql+pytds://user:***@ip/db)
and then I do %load_ext sql
and 
%%sql

Select ...

The result is : Done.
(pytds.tds_base.ProgrammingError) Previous statement didn't produce any results
(Background on this error at: https://sqlalche.me/e/14/f405)

I know the select produces a result if I just do:
import pytds
with pytds.connect('ip, 'db', 'sa', pw as conn:
    with conn.cursor() as cur:
        cur.execute("SELECT ...")
        a = cur.fetchall()
        print(a)

How to get sqlalchemy and the sql extension to work?

**Comment by CaselIT** [comment]
Hi,
 
The `%%sql` jupyter command and pytds are not maintained by sqlalchemy developers so you may have to report the issue to their maintainers.

**Comment by ifuchs** [comment]
I fear I’m going to get into a finger pointing exercise. Perhaps someone
can suggest what is wrong with what I’ve done so far.

**Reply by zzzeek** [reply]
nothing wrong above, code is fine.

**Comment by ifuchs** [comment]
That’s what I thought which is why I don’t understand why I’m not getting
any result. The same SQL statement done without the plug-in gives the
correct  result.

**Reply by zzzeek** [reply]
well that's the kicker here, we don't know either.   it has to do with something called "pytds" that you appear to have installed.   That's also not a mistake!  SQLAlchemy has a lot of [third party dialects](https://docs.sqlalchemy.org/en/14/dialects/index.html#external-dialects) available.     The thing is, we (me and @CaselIT and a few other people here) don't actually write those dialects, or work on them, or have any experience using them, and we don't really even know who made them or what (That's why theyre extensions).

So if we're trying to think who might know the answer to your question, I can say it's not us here!  it's probably the people who wrote "pytds".   Do with that information what you will.

**Comment by ifuchs** [comment]
Thanks. I now doubt it is sqlalchemy as this works fine:

import sqlalchemy
engine = ***@***.***/db')
connection = engine.connect()
result = connection.execute("select * from tbl where column=‘blah'")
for row in result:
    print(row)

However, if I do:
%load_ext sql
%sql ***@***.***:1433/db ***@***.***:1433/db>
select * from tbl where column=‘blah’

I get:
Done.
(pytds.tds_base.ProgrammingError) Previous statement didn't produce any results
(Background on this error at: https://sqlalche.me/e/14/f405 <https://sqlalche.me/e/14/f405>)

I posted to the other lists but haven’t gotten any responses as yet.
